newmodels/server.lua

```diff
@@ -237,74 +313,12 @@ function doModListChecks()
 
 		for k,mod in pairs(mods) do
 
-			-- 1.  verify IDs
-			if not tonumber(mod.id) then
-				return modCheckError("Invalid mod ID '"..tostring(mod.id).."'")
-			else
-				if mod.id == 0 then
-					return modCheckError("Invalid mod ID '"..tostring(mod.id).."', must be >0")
-				end
-
-				if isDefaultID(false, mod.id) then
-					return modCheckError("Invalid mod ID '"..tostring(mod.id).."', must be out of the default GTA:SA and SAMP ID Range, see shared.lua isDefaultID")
-				end
-
-				for _,id in pairs(used_ids) do
-					if id == mod.id then
-						return modCheckError("Duplicated mod ID '"..id.."'")
-					end
-				end
-
-				table.insert(used_ids, mod.id)
-			end
-			if not tonumber(mod.base_id) then
-				return modCheckError("Invalid mod base ID '"..tostring(mod.base_id).."'")
-			else
-				if not isDefaultID(false, mod.base_id) then
-					return modCheckError("Invalid mod base ID '"..tostring(mod.base_id).."', must be a default GTA:SA ID")
-				end
-			end
-
-			-- 2.  verify name
-			if not mod.name or type(mod.name)~="string" then
-
-				return modCheckError("Missing/Invalid mod name '"..tostring(mod.name).."' for mod ID "..mod.id)
-			end
-
-			-- 3.  verify path
-			if (not mod.path) then
-
-				return modCheckError("Missing mod path '"..tostring(mod.path).."' for mod ID "..mod.id)
-			end
-			if not (type(mod.path)=="string" or type(mod.path)=="table") then
-
-				return modCheckError("Invalid mod path '"..tostring(mod.path).."' for mod ID "..mod.id)
-			end
+			local co = coroutine.create(verifyOneMod)
```

Comment:
Using local here defeats the purpose of a coroutine.
You can create a modloader, and have a separate coroutine for each loading mod

```
local mod_routines= {}

for k, mod in pairs(mods) do
    mod_routines[k] = coroutine.create(verifyOneMod)
    coroutine.resume(mod_routines[k], args)
end
```
in the resume function, you let the code run its course. If it is all successful, you can clear the array key, if it's false, you can use coroutine.yield to halt the operation, and call for it again, depending on the use case.